drivers/ethernet/Kconfig.stm32_hal

```diff
@@ -12,6 +12,7 @@ menuconfig ETH_STM32_HAL
 	select NOCACHE_MEMORY if SOC_SERIES_STM32H7X && CPU_CORTEX_M7
 	select HWINFO
 	select ETH_DSA_SUPPORT
+	imply MDIO if SOC_SERIES_STM32H5X || SOC_SERIES_STM32H7X
```

Comment:
Why using imply here ? Ie is that possible that MDIO would not be available if H5 or H7, and if yes would it be  functional anyway ?

Reply:
MDIO hardware and regs are part of ETH IP module, so it's always there when ETH_STM32_HAL is set.

Reply:
See "imply" definition here: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#the-imply-statement.
I really think you should use "select" here.

Reply:
@spectrum70 Can you check on my feedback on this point ?

Reply:
yes., select is what i need. Thanks, fixed.